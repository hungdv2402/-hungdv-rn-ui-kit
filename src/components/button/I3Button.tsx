import React from 'react';
import { Text, StyleProp, StyleSheet, TextStyle, TouchableOpacity, View, ViewStyle } from 'react-native';
import SizeBox from '../sizeBox';

export interface I3ButtonProps {
  icon?: React.ReactNode;
  disabled?: boolean;
  label?: string;
  primaryType?: boolean;
  secondaryType?: boolean;
  textStyle?: StyleProp<TextStyle>;
  buttonStyle?: StyleProp<ViewStyle>;
  thirdType?: boolean;
  onPress(): void;
}

function I3Button(props: I3ButtonProps) {
  const {
    icon,
    disabled,
    label,
    primaryType = true,
    secondaryType,
    thirdType,
    buttonStyle,
    textStyle,
    onPress,
  } = props;

  return (
    <TouchableOpacity
      onPress={onPress}
      disabled={disabled}
      style={[
        primaryType && styles.primary,
        secondaryType && { backgroundColor: '#fff' },
        thirdType && { backgroundColor: '#fff', borderColor: '#F0F0F0' },
        disabled && {
          backgroundColor: '#EBEBEB',
          borderColor: '#EBEBEB',
        },
        buttonStyle,
      ]}
    >
      <View style={styles.titleWrapper}>
        {icon && (
          <>
            {icon}
            <SizeBox width={8} />
          </>
        )}
        <Text
          style={[
            primaryType && styles.textPrimary,
            secondaryType && { color: '#16AB89' },
            thirdType && { color: '#606163' },
            disabled && { color: '#919192' },
            textStyle,
          ]}
        >
          {label}
        </Text>
      </View>
    </TouchableOpacity>
  );
}

export default I3Button;

const styles = StyleSheet.create({
  primary: {
    borderRadius: 28,
    backgroundColor: '#16AB89',
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'stretch',
    height: 56,
    borderColor: '#16AB89',
    borderWidth: 2,
  },
  textPrimary: {
    fontSize: 16,
    color: '#fff',
    fontWeight: '700',
  },
  titleWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
  },
});
