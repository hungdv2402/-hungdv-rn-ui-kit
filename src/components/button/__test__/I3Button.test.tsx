import renderer from 'react-test-renderer';
import I3Button from '../I3Button';
import React from 'react';
import { Image } from 'react-native';

const customIcon = require('../assets/close.png');

describe('renders I3Button correctly', () => {
  it('I3button with primaryType', () => {
    const tree = renderer
      .create(
        <I3Button
          label="Submit"
          onPress={jest.fn()}
          disabled={false}
          primaryType
          textStyle={{ color: 'blue' }}
          buttonStyle={{ borderColor: 'blue', backgroundColor: 'orange' }}
        />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('I3button with secondaryType and Icon', () => {
    const tree = renderer
      .create(
        <I3Button
          label="Submit"
          icon={<Image source={customIcon} />}
          onPress={jest.fn()}
          disabled={false}
          secondaryType
        />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('I3button with thirdType and disabled', () => {
    const tree = renderer
      .create(
        <I3Button
          label="Submit"
          onPress={jest.fn()}
          disabled
          thirdType
        />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});