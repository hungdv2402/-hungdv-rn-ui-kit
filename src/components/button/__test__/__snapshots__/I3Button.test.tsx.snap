// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders I3Button correctly I3button with primaryType 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "orange",
      "borderColor": "blue",
      "borderRadius": 28,
      "borderWidth": 2,
      "height": 56,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "700",
          },
          undefined,
          undefined,
          false,
          {
            "color": "blue",
          },
        ]
      }
    >
      Submit
    </Text>
  </View>
</View>
`;

exports[`renders I3Button correctly I3button with secondaryType and Icon 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#fff",
      "borderColor": "#16AB89",
      "borderRadius": 28,
      "borderWidth": 2,
      "height": 56,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/components/button/assets/close.png",
        }
      }
    />
    <View
      style={
        {
          "height": undefined,
          "width": 8,
        }
      }
    />
    <Text
      style={
        [
          {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "700",
          },
          {
            "color": "#16AB89",
          },
          undefined,
          false,
          undefined,
        ]
      }
    >
      Submit
    </Text>
  </View>
</View>
`;

exports[`renders I3Button correctly I3button with thirdType and disabled 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#EBEBEB",
      "borderColor": "#EBEBEB",
      "borderRadius": 28,
      "borderWidth": 2,
      "height": 56,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "700",
          },
          undefined,
          {
            "color": "#606163",
          },
          {
            "color": "#919192",
          },
          undefined,
        ]
      }
    >
      Submit
    </Text>
  </View>
</View>
`;
